### ISSUE TITLE
Can not create application via UI with avp type ibmsecretsmanager

### ISSUE THREAD

**Issue by philippselle** [IBM/argocd-vault-plugin#211]
**Describe the bug**
Hi - I want to use the AVP with the IBM Secrets Manager, but the application creation via the UI is failing with "Error: Must provide a supported Authentication Type Usage".
Since I want to connect to the AVP_TYPE: ibmsecretsmanager I do not provide an authentication type.

Ways I tried to provide the AVP envs:
a) directly via ENV in the plugin menu of the application UI
b) via a secret.yaml and envFrom annotation in the repo-server yaml

If I connect to the running pod and run the command `argocd-vault-plugin generate ./` it works, since the AVP envs are in place.
`sh-4.4$ env | grep AVP`
`AVP_IBM_INSTANCE_URL=https://xxx.secrets-manager.appdomain.cloud/api`
`AVP_TYPE=ibmsecretsmanager`
`AVP_IBM_API_KEY=xxx`

Full error msg:
> Unable to create application: application spec is invalid: InvalidSpecError: Unable to generate manifests in reponame/: rpc error: code = Unknown desc = `argocd-vault-plugin generate ./` failed exit status 1: Error: Must provide a supported Authentication Type Usage: argocd-vault-plugin generate <path> [flags] Flags: -c, --config-path string path to a file containing Vault configuration (YAML, JSON, envfile) to use -h, --help help for generate -s, --secret-name string name of a Kubernetes Secret containing Vault configuration data in the argocd namespace of your ArgoCD host (Only available when used in ArgoCD) Must provide a supported Authentication Type

Versions used:
AVP: 1.1.1
ArgoCD: 2.0.5 (via RH GitOps Operator)

**To Reproduce**
1. Open the ArgoCD UI
2. Create a new ArgoCD Application and use the type ibmsecretsmanager as AVP_TYPE env

**Comment by jkayani** [comment]
Hi, I looked at the docs from v1.1.1: https://github.com/IBM/argocd-vault-plugin/tree/v1.1.1#ibm-cloud-secrets-manager

`AVP_IBM_INSTANCE_URL=https://xxx.secrets-manager.appdomain.cloud/api` - this env var was only added in a later version (I think `v1.3.0`?). So I'm confused why this works - maybe you also have `VAULT_ADDR` set to this value too?

In that version, you also have to set `AVP_AUTH_TYPE: iam`. In later versions we defaulted to `iam` auth automatically but not in v1.1.1
